specification/specs/search/search/hits/Hit.ts

```diff
@@ -1,21 +1,25 @@
 class Hit<TDocument> {
-  _explanation: Explanation;
-  fields: Dictionary<string, LazyDocument>;
-  highlight: Dictionary<string, string[]>;
-  /** @prop_serializer VerbatimInterfaceReadOnlyDictionaryKeysFormatter`2 */
-  inner_hits: Dictionary<string, InnerHitsResult>;
-  matched_queries: string[];
-  _nested: NestedIdentity;
-  _ignored: string[];
-
   _index: IndexName;
   _id: Id;
-  _shard: string;
-  _node: string;
-  _source: TDocument;
-  _seq_no: long;
-  _primary_term: long;
-  _score: double;
-  _version: long;
-  sort: Array<number | string>;
+
+  _score?: double;
+  _type?: TypeName;
+
+  _explanation?: Explanation;
+  fields?: Dictionary<string, LazyDocument>;
+  highlight?: Dictionary<string, string[]>;
+  /** @prop_serializer VerbatimInterfaceReadOnlyDictionaryKeysFormatter`2 */
+  inner_hits?: Dictionary<string, InnerHitsResult>;
+  matched_queries?: string[];
+  _nested?: NestedIdentity;
+  _ignored?: string[];
+
+  _shard?: string;
+  _node?: string;
+  _routing?: string;
+  _source?: TDocument;
+  _seq_no?: long;
+  _primary_term?: long;
+  _version?: long;
+  sort?: Array<number | string>;
```

Comment:
We should try to be as much specific as possible, is this a `long` or a `double`?

Reply:
both lol should we do `long | double | string` ?

Reply:
Yep! :)